pipeline/train/spm-vocab.sh

```diff
@@ -66,10 +66,18 @@ fi
 vocab_dir=$(dirname "${vocab_src_output}")
 mkdir -p "${vocab_dir}"
 
-zstdmt -dc "${merged_corpus_src}" >"${vocab_dir}/data.src.txt"
-zstdmt -dc "${merged_corpus_trg}" >"${vocab_dir}/data.trg.txt"
-
 if [ "$vocab_split" == "true" ] || [ "$vocab_split" == "True" ]; then
+  # Sample the input corpus to at most 10M sentences to avoid running out of disk space
+  # that's the amount SP uses, so it doesn't make sense to save in disk more than that
+  # after that, remove the tab separator
+  paste \
+    <(zstdmt -dc "${merged_corpus_src}") \
+    <(zstdmt -dc "${merged_corpus_trg}") \
+    | shuf -n "${sample_size}" \
```

Comment:
Does shuf use memory or disk for shuffling?

Reply:
Yes, but I've checked and it seems that the `-n` option is only loading in memory the same as output size, therefore I guess we are fine. 5 or 10M sentence pairs should only be a few GB.